core/src/main/scala/org/apache/spark/rdd/ZippedWithIndexRDD.scala

```diff
@@ -64,8 +64,14 @@ class ZippedWithIndexRDD[T: ClassTag](prev: RDD[T]) extends RDD[(T, Long)](prev)
 
   override def compute(splitIn: Partition, context: TaskContext): Iterator[(T, Long)] = {
     val split = splitIn.asInstanceOf[ZippedWithIndexRDDPartition]
-    firstParent[T].iterator(split.prev, context).zipWithIndex.map { x =>
-      (x._1, split.startIndex + x._2)
+    val parentIter = firstParent[T].iterator(split.prev, context)
```

Comment:
we should add a line of comment saying we don't use Scala's zipWithIndex to avoid overflowing.